src/schemas/xspec.rnc

```diff
@@ -310,10 +310,14 @@ assertion =
 	## assertion succeeds only if the sequence is equal to the one specified with the
 	## href and select attributes or content of the <expect> element.
 	element expect { common-attributes,
-		label, pending-attribute?, test?, selection }
+		label, pending-attribute?, test?, result-type?, selection }
 
 test = attribute test { xpath }
 
+result-type =
+	## Data type of actual result.
+	attribute result-type { xsd:string { minLength = "1" } }
```

Comment:
The value is supposed to be a SequenceType (as in [instance of](https://www.w3.org/TR/xpath-31/#id-instance-of)) but I didn't express the SequenceType pattern more specifically because the `as` attribute doesn't, either.